### ISSUE TITLE
R load_pool from data.frame (data.table) issue with categorical variables

### ISSUE THREAD

**Issue by AdrianAntico** [catboost/catboost#1782]
Problem: 
Testing out the new version (0.26) and load_pool can no longer support cat_features whereas version 0.25.1 and prior version can. Not sure if this is expected behavior (breaking change) or an issue. I am using a data.table like I have previously, with numeric and categorical columns. 

```
TrainPool <- catboost::catboost.load_pool(dataTrain., label = TrainTarget., cat_features = cats, weight = TrainWeightVector)
Error in catboost::catboost.load_pool(dataTrain., label = TrainTarget.,  : 
  Parameter 'cat_features' should be NULL when the pool is constructed from data.frame
```
catboost version: 0.26 (compared to earlier versions)
Operating System: PC / Windows 10
CPU: AMD
GPU: NVIDIA Titan (not sure if relevant yet. I did see the message on requiring a newer driver, just haven't test that yet)

**Comment by Glemhel** [comment]
Hello, AdrianAntico!
Thank you for reporting! This behaviour was added in recent commit, and probably, due to possible confusion while using old code conventions, this will be rolled back not to produce the error as soon as possible.

However, I should warn you about usage of data.frame(and hence data.table) with cat_features parameter. In v0.26, as well as in prior versions, providing data as data.frame leads to **ignoring cat_features** parameter. It is just not passed to `catboost.from_data_frame`, where cat_features are detected by checking whether column is a factor or not. 

So, as a temporary workaround, I recommend you either to convert your `data.table` to `matrix` when passing to `catboost.load_pool` _or_ to manually convert columns you want to be cat_features to factors and not to pass this parameter.

As a next step, we will try to come up with more proper behaviour for this interface and more informative error message.

**Comment by AdrianAntico** [comment]
@Evgueni-Petrov-aka-espetrov I've had success previously as evidenced by the variable importance including categorical variables as highly important. Not sure how they could show up as highly important if they were getting ignored altogether...

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
This is a bug. Prior to 0.26, CatBoost converts cat features to hashes, and then uses these hashes in training.

**Comment by kizill** [comment]
Fixed in 0.26.1
